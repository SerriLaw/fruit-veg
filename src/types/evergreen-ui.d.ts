declare module 'evergreen-ui' {
  interface SelectMenuOption {
    label: string;
    value: string | number;
  }

  interface TextProps {
    size?: number;
  }

  export interface SelectMenuProps {
    selected: string;
    hasTitle: boolean;
    onSelect: (option: SelectMenuOption) => any;
    options?: SelectMenuOption[];
    closeOnSelect?: boolean;
  }
  export class SelectMenu extends React.PureComponent<SelectMenuProps> {}

  export interface IconProps {
    icon: string;
    size?: number;
    title?: string;
    color?: string;
    style?: any;
  }
  export class Icon extends React.PureComponent<IconProps> {}

  interface ButtonProps {
    intent?: 'none' | 'success' | 'danger' | 'warning';
    appearance?: 'primary' | 'minimal';
    iconBefore?: string;
    iconAfter?: string;
    disabled?: boolean;
    className?: string;
    height?: number;
    paddingBottom?: number;
    paddingTop?: number;
    onClick?: (...args: any) => void;
  }
  export class Button extends React.PureComponent<ButtonProps> {}
  export class Heading extends React.PureComponent<TextProps> {}
  export class Paragraph extends React.PureComponent<any> {}
  export class Pane extends React.PureComponent<any> {}
}
