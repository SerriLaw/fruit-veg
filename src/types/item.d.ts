import { Season } from '../lib/enum';

export interface RawItem {
  name: string;
  type: string;
  season: string;
  month: string;
}

export interface Item {
  name: string;
  type: 'fruit' | 'vegetable';
  seasons: Season[];
  months: string[];
}
