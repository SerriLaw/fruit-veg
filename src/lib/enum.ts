export enum Season {
  Summer = 'summer',
  Autumn = 'autumn',
  Winter = 'winter',
  Spring = 'spring'
}
