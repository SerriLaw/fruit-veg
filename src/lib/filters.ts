import { Season } from './enum';
import { Item } from '../types/item';

export function byMonth(items: Item[], month: string): Item[] {
  return items.filter(x => x.months.includes(month));
}

export function bySeason(items: Item[], season: Season): Item[] {
  return items.filter(x => x.seasons.includes(season));
}

export function byType(items: Item[], type: string): Item[] {
  return items.filter(x => x.type === type);
}
