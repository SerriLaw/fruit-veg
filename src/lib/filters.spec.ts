import { byMonth, bySeason, byType } from './filters';
import { Season } from './enum';
import { Item } from '../types/item';

const mockItems: Item[] = [
  {
    name: 'apple',
    type: 'fruit',
    months: ['april', 'may'],
    seasons: [Season.Autumn]
  },
  {
    name: 'banana',
    type: 'fruit',
    months: ['january', 'march'],
    seasons: [Season.Autumn, Season.Summer]
  },
  {
    name: 'carrot',
    type: 'vegetable',
    months: ['december', 'november'],
    seasons: [Season.Autumn, Season.Spring]
  }
];

describe('Filters', () => {
  it('should filter data for given month', () => {
    expect(byMonth(mockItems, 'march')).toMatchSnapshot();
    expect(byMonth(mockItems, 'fake')).toMatchSnapshot();
  });

  it('should filter data for given season', () => {
    expect(bySeason(mockItems, Season.Summer)).toMatchSnapshot();
    expect(bySeason(mockItems, Season.Spring)).toMatchSnapshot();
  });

  it('should filter data for given type', () => {
    expect(byType(mockItems, 'fruit')).toMatchSnapshot();
    expect(byType(mockItems, 'vegetable')).toMatchSnapshot();
  });
});
