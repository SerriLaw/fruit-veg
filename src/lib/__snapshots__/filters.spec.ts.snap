// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters should filter data for given month 1`] = `
Array [
  Object {
    "months": Array [
      "january",
      "march",
    ],
    "name": "banana",
    "seasons": Array [
      "autumn",
      "summer",
    ],
    "type": "fruit",
  },
]
`;

exports[`Filters should filter data for given month 2`] = `Array []`;

exports[`Filters should filter data for given season 1`] = `
Array [
  Object {
    "months": Array [
      "january",
      "march",
    ],
    "name": "banana",
    "seasons": Array [
      "autumn",
      "summer",
    ],
    "type": "fruit",
  },
]
`;

exports[`Filters should filter data for given season 2`] = `
Array [
  Object {
    "months": Array [
      "december",
      "november",
    ],
    "name": "carrot",
    "seasons": Array [
      "autumn",
      "spring",
    ],
    "type": "vegetable",
  },
]
`;

exports[`Filters should filter data for given type 1`] = `
Array [
  Object {
    "months": Array [
      "april",
      "may",
    ],
    "name": "apple",
    "seasons": Array [
      "autumn",
    ],
    "type": "fruit",
  },
  Object {
    "months": Array [
      "january",
      "march",
    ],
    "name": "banana",
    "seasons": Array [
      "autumn",
      "summer",
    ],
    "type": "fruit",
  },
]
`;

exports[`Filters should filter data for given type 2`] = `
Array [
  Object {
    "months": Array [
      "december",
      "november",
    ],
    "name": "carrot",
    "seasons": Array [
      "autumn",
      "spring",
    ],
    "type": "vegetable",
  },
]
`;
